src/Middleware.php

```diff
@@ -114,23 +117,22 @@ protected function validate(Request $request, Closure $next, PathItem $pathItem)
      * @throws IOException
      * @throws InvalidJsonPointerSyntaxException
      */
-    protected function pathItem(string $requestPath, string $requestMethod): PathItem
+    protected function pathItem(Route $route, string $requestMethod): array
     {
-        if (! Str::startsWith($requestPath, '/')) {
-            $requestPath = '/'.$requestPath;
-        }
+        $requestPath = Str::start($route->uri(), '/');
 
         $openapi = $this->spectator->resolve();
 
         $this->version = $openapi->openapi;
 
         foreach ($openapi->paths as $path => $pathItem) {
-            if ($this->resolvePath($path) === $requestPath) {
+            $resolvedPath = $this->resolvePath($path);
+            if (Str::match($route->getCompiled()->getRegex(), $resolvedPath) !== '') {
```

Comment:
Hi @bastien-phi ,

Thanks for your PR. I've applied it in my project and encountered a situation.

If there is an API `PATCH /users/me`, which provides a user to update his/her information and an exception may be thrown at Line 138, because, there is only `patch` method `/users/me` path.

Reply:
Thanks @ymhuang0808 ! Nice catch.
I will look into this

Reply:
@ymhuang0808 I think the changes resolve this issue. Can you try on your project just in case ?

Reply:
@bastien-phi Thanks for your changes. It works for me.